File: leetcode/solutions/1221.SplitAStringInBalancedStrings/balancedStringSplit.go
package solution

func balancedStringSplit(s string) int {
	var (
		start  rune
		result int
		count  int
	)

	for _, r := range s {
		switch {
		case count == 0:
			result++
			count++
			start = r

		case r == start:
			count++

		default:
			count--
		}
	}

	return result
}

File: leetcode/solutions/1221.SplitAStringInBalancedStrings/balancedStringSplit.ts
export default function balancedStringSplit(s: string): number {
  let result = 0;
  let count = 0;

  for (const c of s) {
    count += c === "R" ? 1 : -1;
    if (count === 0) {
      result++;
    }
  }

  return result;
}
